services/worker-manager/src/providers/azure/index.js

```diff
@@ -466,7 +466,13 @@ class AzureProvider extends Provider {
       let asn1 = forge.asn1.fromDer(forge.util.createBuffer(decodedMessage));
       message = forge.pkcs7.messageFromAsn1(asn1);
     } catch (err) {
-      this.monitor.log.registrationErrorWarning({ message: 'Error extracting PKCS#7 message', error: err.toString() });
+      this.monitor.log.registrationErrorWarning({
+        message: 'Error extracting PKCS#7 message',
+        error: err.toString(),
+        workerPoolId: workerPool.workerPoolId,
+        providerId: this.providerId,
+        workerId: worker.workerId,
+      });
```

Comment:
Inside this method (so, inside the scope wehre workerPool, this, and worker are defined), you could define a closure to do this:
```js
   let logRegError = (err, message) => {
      this.monitor.log.registrationErrorWarning({
        message, err,
        workerPoolId: workerPool.workerPoolId,
        providerId: this.providerId,
        workerId: worker.workerId,
      });
```

Another option is to make a child logger that contains some of those values, but then you're passing a reference to that logger everywhere.

TBH I wouldn't worry about the verbosity here.

Reply:
*option 3 - Add a comment and Dustin will prototype the solution* 😁  

Thanks! I'm going to try the closure, see what it looks like.

Reply:
I ran out of time to make this look nice. Maybe when someone re-writes the Azure provisioner they can fix this...